### PR TITLE
Use pbs python api for hosts and jobs

### PR DESCRIPTION
**Pull request by stdweird**

### CODE REVIEW

File: lib/vsc/jobs/pbs/nodes.py [hunk 1]
```diff
@@ -171,12 +171,14 @@ def get_nodes_dict():
             _log.debug('Added down_on_error node %s' % (name))
             states.insert(0, ND_down_on_error)
 
-        if 'jobs' in full_state and not all([JOBID_REG.search(x.strip()) for x in full_state['jobs']]):
-            _log.debug('Added bad node %s for jobs %s' % (name, full_state['jobs']))
-            states.insert(0, ND_bad)
-
         # extend the node dict with derived dict (for convenience)
         derived = {}
+        if 'jobs' in full_state:
```
Comment (boegel) [inline lib/vsc/jobs/pbs/nodes.py]: magic string `'jobs'`?

File: lib/vsc/jobs/pbs/nodes.py [hunk 2]
```diff
@@ -171,12 +171,14 @@ def get_nodes_dict():
...
+            jobs = full_state.get_jobs()
+            if not all([JOBID_REG.search(x.strip()) for x in jobs]):
+                _log.debug('Added bad node %s for jobs %s' % (name, jobs))
+                states.insert(0, ND_bad)
+            derived['jobs']=jobs
```
Comment (boegel) [inline lib/vsc/jobs/pbs/nodes.py]: spaces around ` = `

File: lib/vsc/jobs/pbs/nodes.py [hunk 3]
```diff
@@ -171,12 +171,14 @@ def get_nodes_dict():
...
-        if 'jobs' in full_state and not all([JOBID_REG.search(x.strip()) for x in full_state['jobs']]):
-            _log.debug('Added bad node %s for jobs %s' % (name, full_state['jobs']))
-            states.insert(0, ND_bad)
-
         # extend the node dict with derived dict (for convenience)
         derived = {}
+        if 'jobs' in full_state:
+            jobs = full_state.get_jobs()
+            if not all([JOBID_REG.search(x.strip()) for x in jobs]):
+                _log.debug('Added bad node %s for jobs %s' % (name, jobs))
```
Comment (boegel) [inline lib/vsc/jobs/pbs/nodes.py]: include `ND_bad` in log message (not sure if this makes sense)

**Comment by boegel** [comment]
very minor remarks, looks good overall

File: lib/vsc/jobs/pbs/nodes.py (hunk 1 above)
Comment (stdweird) [inline lib/vsc/jobs/pbs/nodes.py]: well, if you really insist. the whole code is littered with them :smile:

File: lib/vsc/jobs/pbs/nodes.py (hunk 3 above)
Comment (stdweird) [inline lib/vsc/jobs/pbs/nodes.py]: i don't think it makes much sense. `ND_bad` is a indentifier we invented to track bad nodes (the value itself is meaningless)

**Comment by stdweird** [comment]
@boegel remarks processed

File: lib/vsc/jobs/pbs/nodes.py [hunk 4]
```diff
@@ -118,13 +118,21 @@
 
 JOBID_REG = re.compile(r"\w+/\w+(\.|\w|\[|\])+")
 
+ATTR_STATES = 'states'
+ATTR_STATE = 'state'
+ATTR_NODESTATE = 'nodestate'
+ATTR_NP = 'np'
+ATTR_STATUS = 'status'
+ATTR_JOBS = 'jobs'
+ATTR_ERROR = 'error'
```
Comment (boegel) [inline lib/vsc/jobs/pbs/nodes.py]: sort alphabetically?

**Comment by boegel** [comment]
tiny remark, good to go otherwise

**Comment by stdweird** [comment]
@boegel fixed

**Pull request merged**